src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/HealthCheckFunctionTests.cs

```diff
@@ -0,0 +1,54 @@
+using System.Net;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Services.Interfaces;
+using Microsoft.AspNetCore.Http;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests.Functions;
+
+public abstract class HealthCheckFunctionTests(ProcessorFunctionsIntegrationTestFixture fixture)
+    : ProcessorFunctionsIntegrationTest(fixture)
+{
+    public class HealthCheckTests(ProcessorFunctionsIntegrationTestFixture fixture)
+        : HealthCheckFunctionTests(fixture)
+    {
+        [Fact]
+        public async Task Success()
+        {
+            // Ensure that the test folder for simulating the File Share Mount is present prior to
+            // running the Health Check.
+            var dataSetVersionPathResolver = GetRequiredService<IDataSetVersionPathResolver>();
+            Directory.CreateDirectory(dataSetVersionPathResolver.BasePath());
+            
+            var function = GetRequiredService<HealthCheckFunctions>();
+
+            var httpContext = new DefaultHttpContext();
+            var result = await function.HealthCheck(httpContext.Request);
+
+            var expectedHealthCheckResult = new HealthCheckFunctions.HealthCheckResponse(
+                PsqlConnection: new HealthCheckFunctions.HealthCheckSummary(Healthy: true),
+                FileShareMount: new HealthCheckFunctions.HealthCheckSummary(Healthy: true));
+            
+            result.AssertOkObjectResult(expectedHealthCheckResult);
+            Assert.Equal((int) HttpStatusCode.OK, httpContext.Response.StatusCode);
```

Comment:
Is an additional status code check required here? I would have though that the `AssertOkObjectResult` check would be doing that.

Reply:
Yup sorry that was a hangover from previous work - removed now ta